internal/cache/redis.go

```diff
@@ -22,29 +22,54 @@ func NewRedisStore(ctx context.Context, c redis.UniversalClient) *RedisStore {
 	}
 }
 
-// Get retrieves a value by key.
-func (r *RedisStore) Get(key string) ([]byte, error) {
-	result, err := r.client.Get(r.ctx, key).Result()
+// Get retrieves a value by key and hash (if given).
+func (r *RedisStore) Get(hash *string, key string) ([]byte, error) {
+	var result string
+	var err error
+
+	if hash == nil {
+		result, err = r.client.Get(r.ctx, key).Result()
+	} else {
+		result, err = r.client.HGet(r.ctx, *hash, key).Result()
+	}
```

Comment:
Currently, we're only saving and returning a `[]byte` type from the store. One idea for improvement would be to make this store generic and use jsons. Example:

```go
type RedisStore struct {
	// ...
}

func (r *RedisStore) Get[T any](key string) (T, error) {
	val, err := r.client.Get(r.ctx, key)
	if err != nil {
		return result, err
	}

	// Assume Redis values are stored as JSON strings
	var result T
	err = json.Unmarshal([]byte(val), &result)
	if err != nil {
		return result, ErrFailedToParseValue
	}

	return result, nil
}

func (r *RedisStore) Set[T any](key string, value T) error {
	// Marshal the value into JSON
	data, err := json.Marshal(value)
	if err != nil {
		return ErrFailedToMarshall
	}

	err = r.client.Set(r.ctx, key, string(data))
	if err != nil {
		return ErrCacheFailedToSet
	}

	return nil
}
```

Reply:
Maybe I'm going too far and this is not required, or maybe it's for another task, but I'm throwing the idea here.

WDYT? @boecklim @arkadiuszos4chain @pawellewandowski98

Reply:
Hey @kuba-4chain 
Right here I can't see how the generic functions are of much use as there are no 2 functions which are almost the same and a generic function could reduce duplication.

Or if I just not see it, maybe it could be added in a follow-up PR